src/data/db.ts

```diff
@@ -92,28 +92,59 @@
   });
 }
 
-export function updateRoomIdentities(idc: string, roomIds: string[]): Promise<any> {
+export async function updateRoomIdentities(idc: string, roomIds: string[]): Promise<any> {
   return prisma.rooms
     .findMany({
       where: { id: { in: roomIds } }
     })
     .then((rooms) => {
-      const roomsToUpdate = rooms
-        .filter((room) => !room.identities.includes(idc))
+      const identityListRooms = rooms
+        .filter((room) => room.membershipType === "IDENTITY_LIST" && !room.identities.includes(idc))
         .map((room) => room.id);
 
-      if (roomsToUpdate) {
+      if (identityListRooms.length > 0) {
         return prisma.rooms.updateMany({
-          where: { id: { in: roomsToUpdate } },
+          where: { id: { in: identityListRooms } },
           data: { identities: { push: idc } }
         });
       }
+      const bandadaGroupRooms = rooms
+      .filter((room) => room.membershipType === "BANDADA_GROUP" && !room.identities.includes(idc))
+      .map((room) => room);
+
+      if (bandadaGroupRooms.length > 0) {
+          bandadaGroupRooms.forEach((room) => {
+            if (!room.bandadaAPIKey) {
+              console.error("API key is missing for room:", room);
+              return;
+            }
+
+            const requestOptions = {
+              method: 'POST',
+              headers: {
+                'Content-Type': 'application/json',
+                'x-api-key': room.bandadaAPIKey,
+              },
+            };
+
+            const url = `https://api.bandada.pse.dev/groups/${room.bandadaAddress}/members/${idc}`;
+
+            fetch(url, requestOptions)
```

Comment:
@codetrauma we should try to turn the `idc` into a bigint first, and then back to a string and throw an error if you can't convert it into a bigint. This will help sus out someone trying to do SQL injection or fuzzing attacks.